src/layer/XYZ.ts

```diff
@@ -9,6 +9,12 @@ import {Material} from "./Material";
 import type {NumberArray4} from "../math/Vec4";
 import type {FetchCache, IResponse} from "../utils/Loader";
 
+declare global {
+  interface Window {
+    nsGmx: any;
```

Comment:
This code breaks the modularity and isolation principles of this module, and introduces an undesirable dependency on an external context nsGmx.

Specifically, the XYZ module, which is meant to represent a generic, reusable layer, should not be aware of or depend on entities or names like nsGmx, which are specific to the Geomixer environment. This creates tight coupling and makes the module less portable and harder to maintain.

Recommendation: Override or extend XYZ in your module, and add the nsGmx-specific type or logic in a separate module that is clearly scoped to your environment. This keeps the core module clean, decoupled, and reusable.